BaseStation/MainWindow/MapTileDownloader.cs

```diff
@@ -0,0 +1,175 @@
+            {
+                String size;
+                parameters.TryGetValue("size", out size);
+                return size;
+            }
+
+            // adds the scale to the current configuration
+            public void SetImageScale(int scale)
+            {
+                if (parameters.ContainsKey("scale")) parameters.Remove("scale");
+                parameters.Add("scale", scale.ToString());
+            }
+
+            // returns the scale of the image in the configuration
+            public String GetScale()
+            {
+                String scale;
+                parameters.TryGetValue("scale", out scale);
+                return scale;
+            }
+
+            // adds the coords to the current confuguration
+            public void SetLocation(Tuple<double, double> coords)
+            {
+                if (parameters.ContainsKey("center")) parameters.Remove("center");
+                parameters.Add("center", coords.Item1 + "," + coords.Item2);
+            }
+
+            // returns the location of the center for the current config
+            public String GetLocation()
+            {
+                String center;
+                parameters.TryGetValue("center", out center);
+                return center;
+            }
+
+            // adds the zoom level to the current configuration
+            public void SetZoom(int zoom)
+            {
+                if (parameters.ContainsKey("zoom")) parameters.Remove("zoom");
+                parameters.Add("zoom", zoom.ToString());
+            }
+
+            // returns the zoom value of the configuration
+            public String GetZoom()
+            {
+                String zoom;
+                parameters.TryGetValue("zoom", out zoom);
+                return zoom;
+            }
+
+            // returns the string representation of the configuration
+            public override String ToString()
+            {
+                String result = "";
+                foreach (String key in parameters.Keys)
+                {
+                    result += key + "=" + parameters[key] + "&";
+                }
+                result += "key=" + Environment.GetEnvironmentVariable("GOOGLEMAPSAPIKEY");
+                return result;
+            }
+        }
+
+        // gets the tile set of maps with the given coords of the center, width and height of tiling
+        // and configuration for the center tile
+        public static void DownloadNewTileSet(Tuple<int, int> tilingDim, Configuration config, String mapSetName)
+        {
+            String fileName = Directory.GetCurrentDirectory().ToString() + @"\MapTiles\" + mapSetName + ".txt";
+            using (StreamWriter file = new StreamWriter(fileName))
+            {
+                int zoom;
+                int imgWidth;
+                int imgHeight;
+                double longe;
+                double lat;
+
+                String[] imgDim = config.GetImageDim().Split('x');
```

Comment:
Can the dimensions be added to the configuration object so we don't have to extract the information from a string?